src/Raven.Server/Web/System/OngoingTasksHandler.cs

```diff
@@ -784,9 +784,36 @@ public async Task PutConnectionString()
                             SecurityClearanceValidator.AssertSecurityClearance(JsonDeserializationClient.OlapConnectionString(readerObject), feature?.Status);
                             break;
                     }
+                    
+                    List<string> errors = new ();
+                    if (GetConnectionString(readerObject, connectionStringType).Validate(ref errors) == false)
+                    {
+                        throw new InvalidDataException(string.Join("\n", errors));
+                    }
                 });
         }
 
+        private static ConnectionString GetConnectionString(BlittableJsonReaderObject readerObject, ConnectionStringType connectionStringType)
+        {
+            switch (connectionStringType)
+            {
+                case ConnectionStringType.Raven:
+                    return JsonDeserializationCluster.RavenConnectionString(readerObject); 
+                case ConnectionStringType.Sql:
+                    return JsonDeserializationCluster.SqlConnectionString(readerObject);
+                case ConnectionStringType.Olap:
+                    return JsonDeserializationCluster.OlapConnectionString(readerObject);
+                case ConnectionStringType.ElasticSearch:
+                    return JsonDeserializationCluster.ElasticSearchConnectionString(readerObject);
+                case ConnectionStringType.Queue:
+                    return JsonDeserializationCluster.QueueConnectionString(readerObject);
+                case ConnectionStringType.None:
+                default:
+                    throw new ArgumentOutOfRangeException();
```

Comment:
Missing `connectionStringType` value. We won't know what value we got

Reply:
fixed